File: BE/main_service/delivery/http/admin/facility_registration/facility_registration.go
package facility_registration

import (
	"main_service/endpoints/admin/facility_registration"
	"main_service/model"
	"main_service/util"

	"github.com/gin-gonic/gin"
)

// SetupRouter .
func SetupRouter(parrentRoute gin.IRouter, path string, endpoints facility_registration.Endpoints) {
	route := parrentRoute.Group(path)
	route.GET("/", util.ServeHttpForGin(endpoints.GetAll, &model.FacilityRegistration{}))
	route.PUT("/:id/reject", util.ServeHTTPForGin(
		DecodeRejectFacilityRegistration,
		endpoints.Reject,
	))
	route.PUT("/:id/accept", util.ServeHTTPForGin(
		DecodeAcceptFacilityRegistration,
		endpoints.Accept,
	))
}

File: web/main/api/facilityRegistration.js
import http from '../utils/http'

export function getFacilityRegistrations() {
    return http.get(`/admin/facility-registration/`)
}

export function accept(id) {
    return http.put(`/admin/facility-registration/${id}/accept`)
}

export function reject(id) {
    return http.put(`/admin/facility-registration/${id}/reject`)
}